### PR TITLE
Combine different release pipelines by the use of reusable workflows

### PR DESCRIPTION
**Pull request by andife**
### Description

This pull request should be the start of a pipeline (current status of the considerations under https://github.com/onnx/onnx/pull/6246) that automatically generates a whls after tagging, signs it and publishes it directly to Pypi to improve supply chain security. As this is all too much at once, the changes will be introduced and tested gradually. 

Changes related to this PR:
Reusable Workflows are used to merge the individual OS release builds together (we need one process as we want to have all artifacts in one pipeline, so there is not need to download them manuelly). The workflows are currently not really "reusable", as we more or less use pypi creditionals in the workflow, among other things. I reused our testpypi weekly repo as I didn't want to change our behavoir for pypi and testpypi.

Maybe it would make sense to adapt the process right away so that you can trigger our "Weekly" manually via the Github interface for testing and don't have to wait a week ;-)

#### Key points for the review:
* How do we want to configure concurrency
* What concurrency-groups do we need?
* Where do we need "cancel-in-progress"?

#### Not part of the pull request:
* Extract publish whl to pypi/testpyp
* Introduce trusted publishing
* ... 

### Motivation and Context
* see https://github.com/onnx/onnx/pull/6246

### CODE REVIEW

File: .github/workflows/create_release.yml [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Caller Workflow
+on:
```
Comment (justinchuby) [inline .github/workflows/create_release.yml:6]:
```suggestion
on:
```

nit: trailing spaces.

could you format all new yml files with an auto formatter? Thanks!

**Review by justinchuby — APPROVED** [review]

**Review by justinchuby — CHANGES_REQUESTED** [review]
@andife For the create release workflow, is it possible to skip the publish step in pull requests? Should we only allow it when the trigger is schedule?

File: .github/workflows/create_release.yml [hunk 2]
```diff
@@ -0,0 +1,86 @@
+# Copyright (c) ONNX Project Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+name: Caller Workflow
```
Comment (justinchuby) [inline .github/workflows/create_release.yml:5]:
```suggestion
name: Create Releases
```

nit: suggested name

**Comment by andife** [comment]
> @andife For the create release workflow, is it possible to skip the publish step in pull requests? Should we only allow it when the trigger is schedule?

Both variants would be ok for me.  

I have added conditions in this regard. Could they be simplified to really only execute it on schedule? On the other hand, you might still have the download step separate? 

I at schedule we are creating a preview build. I am convinced that it also makes sense to be able to trigger a preview build manually and that the packages are then also pushed to https://pypi.org/project/onnx-weekly/ with the current date. But this I will/would not include into this PR

File: .github/workflows/create_release.yml (hunk 1 above)
Comment (andife) [reply .github/workflows/create_release.yml]: Done

**Review by justinchuby — APPROVED** [review]

**Comment by andife** [comment]
**Note for clarification: Trusted Publishing does not work for forks (as wanted)**
```

Error: Trusted publishing exchange failure: 
OpenID Connect token retrieval failed: GitHub: missing or insufficient OIDC token permissions, the ACTIONS_ID_TOKEN_REQUEST_TOKEN environment variable was unset

The workflow context indicates that this action was called from a
pull request on a fork. GitHub doesn't give these workflows OIDC permissions,
even if `id-token: write` is explicitly configured.

To fix this, change your publishing workflow to use an event that
forks of your repository cannot trigger (such as tag or release
creation, or a manually triggered workflow dispatch).
```

File: .github/workflows/release_win.yml [hunk 3]
```diff
@@ -79,15 +72,15 @@ jobs:
         }
         python -m build --wheel
         Get-ChildItem -Path dist/*.whl | foreach {python -m pip install --upgrade $_.fullname}
-
+
```
Comment (justinchuby) [inline .github/workflows/release_win.yml:75]:
```suggestion

```

**Pull request merged**